src/pip/_internal/cli/base_command.py

```diff
@@ -134,6 +135,14 @@ def main(self, args):
             user_log_file=options.log,
         )
 
+        if sys.version_info[:2] == (3, 4):
+            deprecated(
+                "Python 3.4 support has been deprecated and will be dropped "
+                "in next version. Please upgrade your Python.",
+                replacement=None,
+                gone_in='19.1',
```

Comment:
```suggestion
                gone_in='19.2',
```

From the [Deprecation Policy](https://pip.pypa.io/en/latest/development/release-process/#deprecation-policy):

> Any change to pip that removes or significantly alters user-visible behavior that is described in the pip documentation will be deprecated for a minimum of 6 months before the change occurs.